src/libs/Network/processRequest.js

```diff
@@ -47,28 +48,58 @@ export default function processRequest(request) {
+                NetworkEvents.getLogger().hmmm('[Network] Error: iOS Load Failed error', {message: error.message, status: error.status});
+            } else if (error.message === CONST.ERROR.SAFARI_CANNOT_PARSE_RESPONSE) {
+                // Another cryptic Apple error message. Unclear why this can happen, but some speculation it can be fixed by a browser restart.
+                NetworkEvents.getLogger().hmmm('[Network] Error: Safari "cannot parse response"', {message: error.message, status: error.status});
+            } else if (error.message === CONST.ERROR.GATEWAY_TIMEOUT) {
+                // This error seems to only throw on dev when localhost:8080 tries to access the production web server. It's unclear whether this can happen on production or if
+                // it's a sign that the web server is down.
+                NetworkEvents.getLogger().hmmm('[Network] Error: Gateway Timeout error', {message: error.message, status: error.status});
+            } else if (request.command === 'Push_Authenticate') {
+                // Push_Authenticate requests can return with fetch errors and no message. It happens because we return a non 200 header like 403 Forbidden.
+                // This is common to see if we are subscribing to a bad channel related to something the user shouldn't be able to access. There's no additional information
+                // we can get about these requests.
+                NetworkEvents.getLogger().hmmm('[Network] Error: Push_Authenticate', {message: error.message, status: error.status});
+            } else if (error.message === CONST.ERROR.EXPENSIFY_SERVICE_INTERRUPTED) {
+                // Auth (database connection) is down or bedrock has timed out while making a request. We currently can't tell the difference between these two states.
+                NetworkEvents.getLogger().hmmm('[Network] Error: Expensify service interrupted or timed out', {type: error.type, title: error.title, jsonCode: error.jsonCode});
             } else {
-                // If we get any error that is not "Failed to fetch" create GitHub issue so we can handle it. These requests will not be retried.
-                NetworkEvents.getLogger().alert(`${CONST.ERROR.ENSURE_BUGBOT} unknown error caught while processing request`, {
+                // If we get any error that is not known log an alert so we can learn more about it and document it here.
+                NetworkEvents.getLogger().alert(`${CONST.ERROR.ENSURE_BUGBOT} unknown error caught while processing request - ${error.message}`, {
                     command: request.command,
-                    error: error.message,
+                    message: error.message,
+                    status: error.status,
                 });
             }
 
-            // If we did not throw and we have a persisted request that was cancelled or for an unknown error remove it so it is not retried
-            if (persisted) {
```

Comment:
Are you sure we don't want this for other unknown errors? Said differently, I see that you're only removing persisted requests for `REQUEST_CANCELLED` errors, but no others. Why did you choose that error only?

Is it because the other errors suggest that the request might be "retryable"?

Reply:
>I see that you're only removing persisted requests for REQUEST_CANCELLED errors, but no others. Why did you choose that error only?

Because this is the only error that truly should never be retried.

For the rest of them, if they are destined to fail the worst case is we'll retry them 10 times and they'll fail. 

We are looking into more specific retry logic for the Network Improvements doc so I expect this "retry everything" strategy will be improved on in the future. 

I initially thought it would be a good idea to avoid retrying things that failed for unexplained reasons, but after looking into all these different errors I didn't come across any that obviously shouldn't be retried. That makes me think the default should be to retry everything vs. only retry the knowable things. More or less... my first instinct was wrong here and I am just putting things back while the way they were while we improve the retry system.